Add neutron rehearsal-1 network (#391)
cosmos/testnets: replicated-security/neutron-rehearsal-1/README.md

> # `neuntron-rehearsal-1` Chain Details

Reviewer: ```suggestion
# `neutron-rehearsal-1` Chain Details
```

The `neuntron-rehearsal-1` chain will be launched as a Neutron persistent chain to test Interchain Security functionality.

> * **Chain-ID**: `neuntron-rehearsal-1`

Reviewer: ```suggestion
* **Chain-ID**: `neutron-rehearsal-1`
```

* **SHA256 for genesis file _without CCV state_**: `e5a5e9f8ba69a846d18737c2d2a6a6de0250f45c53c07df274dc80ab9bda2ba0`

* Genesis file hash
  * The SHA256 is used to verify against the genesis file (without CCV state) that the proposer has made available for review.
  * The `neutron-rehearsal-1-genesis-without-ccv.json` file cannot be used to run the chain: it must be updated with the CCV (Cross Chain Validation) state after the spawn time is reached.
  * The genesis file includes funds for a relayer and a faucet account as well as account with funds for different internal needs, `signed_blocks_window` has been set to `864000`, and `min_signed_per_window` has been set to `5%`.
* Binary hash
  * The `neutrond-linux-amd64` binary is only provided to verify the SHA256. It was built with Interchain Security release [`v0.4.0`](https://github.com/neutron-org/neutron/releases/tag/v0.4.0). You can generate the binary following the build instructions in that repo or using one of the scripts provided here.
* Spawn time
  * Even if a proposal passes, the CCV state will not be available from the provider chain until after the spawn time is reached.

> > If you want to build genesis by your own please take into account difference between `ccvconsumer` section generated by current version of `gaiad` and `ccvconsumer` required for neutron to run. Please check and add `"soft_opt_out_threshold": "0.05"` to the `params` section if it absent.

Reviewer: ```suggestion
> If you want to build genesis by your own please take into account difference between the `ccvconsumer` section generated by version v9.0.2 of `gaiad` and the `ccvconsumer` section required for neutron to run. Please check and add `"soft_opt_out_threshold": "0.05"` to the `params` section if it absent.
```

For more information regarding the consumer chain creation process, see [CCV: Overview and Basic Concepts](https://github.com/cosmos/ibc/blob/main/spec/app/ics-028-cross-chain-validation/overview_and_basic_concepts.md).

## Endpoints

Endpoints are exposed as subdomains for the sentry and snapshot nodes (described below) as follows:

* `https://rpc.rehearsal.ntrn.tech`
* `https://grpc.rehearsal.ntrn.tech`
* `https://rest.rehearsal.ntrn.tech`

Seed nodes:

1. `e2c07e8e6e808fb36cca0fc580e31216772841df@p2p.rehearsal.ntrn.tech:26656`

The following state sync node serve snapshots every 1000 blocks:

1. `https://rpc.rehearsal.ntrn.tech:443`

## IBC Information

Connections and channels will be posted here shortly after the chain launches.

### Clients

* `07-tendermint-0`
  * Counterparty: [`provider`](/replicated-security/provider/README.md) `07-tendermint-1`

### Channels

* `channel-0`: consumer port
  * Counterparty: [`provider`](/replicated-security/provider/README.md) `channel-2`
* `channel-1`: transfer port

>   * Counterparty: [`provider`](/replicated-security/provider/README.md) `channel-3`

Reviewer: You can remove these lines, I'll update this section after the chain becomes interchain-secured.

## How to Join

### Hardware Requirements

* 4 Cores
* 32 GB RAM
* 2x512 GB SSD

### Software Versions

| Name               | Version  |
|--------------------|----------|
| Neutron            | v0.4.0   |
| Go                 | > 1.18   |

The scripts provided in this repo will install Neutron and set up a Cosmovisor service with the auto-download feature enabled on your machine.

#### Bash Script

The script provided in this repo will install `neutrond` and set up a Cosmovisor service on your machine. 

> Run script provided to set up a `neuntron-rehearsal-1` service:
> * `join-rs-neuntron-rehearsal-1.sh` will create a `neuntron-rehearsal-1` service with `cosmovisor` support.

Reviewer: ```suggestion
Run script provided to set up a `neutron-rehearsal-1` service:
* `join-rs-neutron-rehearsal-1.sh` will create a `neutron-rehearsal-1` service with `cosmovisor` support.
```
